### PR TITLE
Add languages plugin category for setupWizard

### PR DESCRIPTION
**Pull request by LinuxSuRen**
Proposed changelog entries:

* RFE - Add Languages plugin category to the Setup Wizard. Jenkins will be now selecting recommended plugins depending on the browser locale

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
With the current setting it will install the plugin by-default on each instance, I believe that the "suggested" flag should be removed.

BTW, the request does seem to comply with https://github.com/jenkinsci/jep/blob/master/jep/2/README.adoc , so I am fine with adding the category.

Translation assistance could be added here as well once it is cleaned up

**Comment by LinuxSuRen** [comment]
I think it's compatible because the Chinese translation was in the core Jenkins, once we remove them from the core Jenkins than users will get a non-Chinese version.
At least, I could try to decide this from the browser language settings.

**Comment by oleg-nenashev** [comment]
> At least, I could try to decide this from the browser language settings

That would be an interesting experiment with Installation wizard

File: war/src/main/js/api/pluginManager.js [hunk 1]
```diff
@@ -40,6 +40,12 @@ exports.init = function(handler) {
 					plugins.names.push(pluginName);
 					if (plugin.suggested) {
 						plugins.recommendedPlugins.push(pluginName);
+					} else if (pluginCategory.category === "Languages"){
```
Comment (oleg-nenashev) [inline war/src/main/js/api/pluginManager.js]: Tabs vs. spaces

**Review by oleg-nenashev — APPROVED** [review]

**Comment by daniel-beck** [comment]
Have some concerns about this, would like to onhold.

Also, we started with a big process around which plugins are suggested; this seems very different, perhaps too different. Unsure though TBH.

**Comment by LinuxSuRen** [comment]
If move all Chinese localization resource files into a plugin, I think the plugin should be suggested. So I could get your point about this. And the suggestion is dynamic according user-agent.

**Review by jglick — APPROVED** [review]
Seems reasonable to me.

File: core/src/main/resources/jenkins/install/platform-plugins.json [hunk 2]
```diff
@@ -101,5 +101,12 @@
             { "name": "publish-over-ssh" },
             { "name": "ssh" }
         ]
+    },
+    {
+        "category":"Languages",
+        "plugins": [
+            { "name": "locale"},
+            { "name": "localization-zh-cn"}
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply core/src/main/resources/jenkins/install/platform-plugins.json]: Just because of all other plugins' name use `-` instead of `_`.

**Comment by oleg-nenashev** [comment]
@daniel-beck Is it still on hold? I believe this is a good thing to experiment with, no negative feedback so far.
A;so https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/jenkinsci-dev/4WlxJUXnNE4

**Comment by daniel-beck** [comment]
Seems reasonable; interactive testing also seemed to show this works nicely, even when accepting the default set.

**Comment by LinuxSuRen** [comment]
I wonder if we could continue this?

**Comment by oleg-nenashev** [comment]
Nobody voted against so 🚢 🇮🇹

**Comment by oleg-nenashev** [comment]
Added the changelog.

@LinuxSuRen I believe https://github.com/jenkinsci/jep/blob/master/jep/216/README.adoc needs to be updated in order to document this feature and the plugin naming standard.

**Comment by LinuxSuRen** [comment]
@oleg-nenashev Got it. I am going to do this as soon as possible.

**Pull request merged**